jupyterlab_automl/jupyterlab_automl/handlers.py

```diff
@@ -60,109 +95,141 @@ def create_automl_parent(client):
-                    chart_info.append({"name":stat.value, "amount": stat.count})
+    type_summary = defaultdict(int)
+    for column_spec in client.list_column_specs(table_spec.name):
+        if column_spec.data_type.type_code in column_types.keys():
+            type_code = column_types[column_spec.data_type.type_code]
         else:
             type_code = "Unrecognized"
-        summary[type_code] += 1
+        detail_panel = get_detail_panel(column_spec, table_spec.row_count - column_spec.data_stats.null_value_count)
+        type_summary[type_code] += 1
         column_specs.append(
             {
                 "id": column_spec.name,
                 "dataType": type_code,
                 "displayName": column_spec.display_name,
                 "distinctValueCount": column_spec.data_stats.distinct_value_count,
-                "invalidValueCount": rowCount - column_spec.data_stats.valid_value_count,
-                "nullValueCount": str(column_spec.data_stats.null_value_count) + " (" + str(math.floor(column_spec.data_stats.null_value_count / rowCount * 100)) + "%)",
+                "invalidValueCount": table_spec.row_count
+                - column_spec.data_stats.valid_value_count,
+                "nullValueCount": str(column_spec.data_stats.null_value_count)
+                + " ("
+                + str(
+                    math.floor(
+                        column_spec.data_stats.null_value_count
+                        / table_spec.row_count
+                        * 100
+                    )
+                )
+                + "%)",
                 "nullable": column_spec.data_type.nullable,
-                "chartInfo": chart_info,
+                "detailPanel": detail_panel,
             }
         )
-    chart_summary = []
-    for key, val in summary.items():
-        chart_summary.append({"name": key, "amount": val})
-    return column_specs, chart_summary
+    columns_summary = []
+    for key, val in type_summary.items():
+        columns_summary.append({"name": key, "Number of Instances": val})
+    return column_specs, columns_summary
 
 
 def get_table_specs(client, datasetId):
     table_specs = []
     for table_spec in client.list_table_specs(datasetId):
-        column_specs, chart_summary = get_column_specs(client, table_spec.name, table_spec.row_count)
         table_specs.append(
             {
                 "id": table_spec.name,
                 "rowCount": table_spec.row_count,
                 "validRowCount": table_spec.valid_row_count,
                 "columnCount": table_spec.column_count,
-                "columnSpecs": column_specs,
-                "chartSummary": chart_summary,
+                "columnSpecs": get_column_specs(client, table_spec)[0],
+                "chartSummary": get_column_specs(client, table_spec)[1],
             }
-        ) 
-    return {
-        "tableSpecs": table_specs
-    }
+        )
```

Comment:
```
columSpecs, chartSummary = get_column_specs(client, table_spec)
...
{
"columnSpecs": columnSpecs,
"chartSummary": chartSummary
}
```
I think we should try to call to get_column_specs only once

Reply:
nice catch i thought i had removed this...